app/src/main/java/com/framgia/wsm/screen/requestoff/RequestOffViewModel.java

```diff
@@ -685,77 +685,243 @@ private double getSumDateOffHaveSalary() {
-            mRequestDayOffTypesAttribute.setSpecialDayOffSettingId(TypeOfDays.MATERNTY_LEAVE);
-            mRequestDayOffTypesAttribute.setNumberDayOff(mRequestOff.getMaternityLeave());
-            mRequestDayOffTypesAttribute.setDestroy(FALSE);
-            mRequestDayOffTypesAttributes.add(mRequestDayOffTypesAttribute);
+        mRequestDayOffTypesAttributes.add(mRequestDayOffTypesAttribute);
+    }
+
+    private void setAttributeLeaveForCareOfSickChild() {
+        mRequestDayOffTypesAttribute = new OffRequest.RequestDayOffTypesAttribute();
+        mRequestDayOffTypesAttribute.setId(null);
+        mRequestDayOffTypesAttribute.setSpecialDayOffSettingId(
+                TypeOfDays.LEAVE_FOR_CARE_OF_SICK_CHILD);
+        mRequestDayOffTypesAttribute.setNumberDayOff(mRequestOff.getLeaveForCareOfSickChild());
+        mRequestDayOffTypesAttribute.setDestroy(FALSE);
+        if (mRequestOff != null && (mRequestOff.getRequestDayOffTypes() != null
+                || mRequestOff.getRequestDayOffTypes().size() != 0)) {
+            for (int i = 0; i < mRequestOff.getRequestDayOffTypes().size(); i++) {
+                if (mRequestOff.getRequestDayOffTypes().get(i).getSpecialDayOffSettingId()
+                        == Integer.parseInt(
+                        RequestOffViewModel.TypeOfDays.LEAVE_FOR_CARE_OF_SICK_CHILD)) {
+                    mRequestDayOffTypesAttribute.setId(
+                            mRequestOff.getRequestDayOffTypes().get(i).getId());
+                }
+            }
+            if (mRequestOff.getLeaveForCareOfSickChild() == null
+                    || "".equals(mRequestOff.getLeaveForCareOfSickChild())
+                    || Constant.DEFAULT_DOUBLE_VALUE.equals(
+                    mRequestOff.getLeaveForCareOfSickChild())) {
+                mRequestDayOffTypesAttribute.setDestroy(TRUE);
+            }
         }
-        if (mRequestOff.getLeaveForCareOfSickChild() != null && !"".equals(
-                mRequestOff.getLeaveForCareOfSickChild())) {
-            mRequestDayOffTypesAttribute.setSpecialDayOffSettingId(
-                    TypeOfDays.LEAVE_FOR_CARE_OF_SICK_CHILD);
-            mRequestDayOffTypesAttribute.setNumberDayOff(mRequestOff.getLeaveForCareOfSickChild());
-            mRequestDayOffTypesAttribute.setDestroy(FALSE);
-            mRequestDayOffTypesAttributes.add(mRequestDayOffTypesAttribute);
+        mRequestDayOffTypesAttributes.add(mRequestDayOffTypesAttribute);
+    }
+
+    private void setAttributeFuneralLeave() {
+        mRequestDayOffTypesAttribute = new OffRequest.RequestDayOffTypesAttribute();
+        mRequestDayOffTypesAttribute.setId(null);
+        mRequestDayOffTypesAttribute.setSpecialDayOffSettingId(TypeOfDays.FUNERAL_LEAVE);
+        mRequestDayOffTypesAttribute.setNumberDayOff(mRequestOff.getFuneralLeave());
+        mRequestDayOffTypesAttribute.setDestroy(FALSE);
+        if (mRequestOff != null && (mRequestOff.getRequestDayOffTypes() != null
```

Comment:
ERROR: [BrokenNullCheck] Method call on object which may be null
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/basic.html#BrokenNullCheck